extensions/panache/mongodb-panache-common/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/BasePanacheMongoResourceProcessor.java

```diff
@@ -381,8 +381,8 @@ protected void processTypes(CombinedIndexBuildItem index,
     }
 
     @BuildStep
-    ReflectiveClassBuildItem registerForReflection() {
-        return new ReflectiveClassBuildItem(true, true, OBJECT_ID.toString());
+    ReflectiveHierarchyIgnoreWarningBuildItem ignoreBsonTypes() {
+        return new ReflectiveHierarchyIgnoreWarningBuildItem(dotname -> dotname.packagePrefix().startsWith("org.bson"));
```

Comment:
I don't remember the specifics but I had some issue with packagePrefix(). I would recommend to do `dotname.toString().startsWith("org.bson.")` (with a dot at the end). Also probably a good idea to make the string a constant.

Reply:
Done